### PR TITLE
Fix test suite

### PR DESCRIPTION
**Pull request by jfmengels**
There were problems running `jest`, and this PR fixes that.

The main problems were the way that `ts-jest` was referenced in the configuration, and then there were some problems about TypeScript not knowing some of the types (fixed by adding `@types/jest`).
I used the opportunity to bump Jest and co to v27.

There were some problems about functions being misused, which I have fixed in a consistent but maybe not thoughtful enough way.

~Tests still don't pass entirely, because the expected/actual are wrong in some of the tests. What is weird is that all the failing tests pass when they are run individually, so there must be some sneaky mutation that impacts all the tests, though in a seemingly consistent behavior. I'd love to help with the remaining tests, but I may look into this further at a later point.~

Tests are passing now. I restrained Jest to only run on TS files. Jest was running both the test/xyz.ts version of the test files and the compiled dist/test/xyz.js version of the test. Removing this duplication also ended up making the tests pass.

I have a followup PR (https://github.com/jfmengels/elm-optimize-more/pull/1) to run tests in GitHub Actions.

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -1,8 +1,7 @@
 function createJestConfig(_, rootDir = __dirname) {
   const config = {
     transform: {
-      '.(ts|tsx)$': require.resolve('ts-jest/dist'),
-      '.(js|jsx)$': require.resolve('babel-jest'),
```
Comment (jfmengels) [inline jest.config.js:5]: Doesn't seem to be necessary (anymore?)

File: test/inlineWrappedFunctions.test.ts [hunk 2]
```diff
@@ -35,7 +35,7 @@ test('it can process nested calls of A2 with non identifiers as the first arg ',
   const { actual, expected } = transformCode(
     initialCode,
     expectedOutputCode,
-    createFunctionInlineTransformer(false, 'for tests')
+    createFunctionInlineTransformer(false, false, 'for tests')
```
Comment (jfmengels) [inline test/inlineWrappedFunctions.test.ts:38]: There was a missing argument for all of these. I just added `false`, but maybe we want some tests with `true` instead (though using `true` also makes the tests pass :man_shrugging: )

File: test/passUnwrappedFunctionsTransformer.test.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { transformCode } from './inlineWrappedFunctions.test';
```
Comment (jfmengels) [inline test/passUnwrappedFunctionsTransformer.test.ts:1]: Importing another test file forced Jest to run the tests inside `inlineWrappedFunctions`, even if you only wanted to run tests for `passUnwrappedFunctionsTransformer`. Also caused the tests from `inlineWrappedFunctions` to be run multiple times (in both test suites).

File: test/inlineWrappedFunctions.test.ts (hunk 2 above)
Comment (mdgriffith) [reply test/inlineWrappedFunctions.test.ts:39]: Nice!  yeah, the tests are super old

**Pull request merged**